src/Fields/Media.php

```diff
@@ -139,6 +146,52 @@ private function setOrder($ids)
         $mediaClass::setNewOrder($ids);
     }
 
+    private function addExistingMedia(NovaRequest $request, $data, HasMedia $model, string $collection, Collection $medias): Collection
```

Comment:
I'd love it if you would extract this ability of the Media field to a designated trait, someting like `HandlesConversionsTrait` or `HandlesCustomPropertiesTrait`.